src/rdiff_backup/Rdiff.py

```diff
@@ -34,8 +34,12 @@ def get_delta_sigrp_hash(rp_signature, rp_new):
     """Like above but also calculate hash of new as close() value"""
     log.Log("Getting delta (with hash) of file {fi} with signature {si}".format(
         fi=rp_new, si=rp_signature), log.DEBUG)
-    return librsync.DeltaFile(
-        rp_signature.open("rb"), hash.FileWrapper(rp_new.open("rb")))
+    try:
+        return librsync.DeltaFile(
+            rp_signature.open("rb"), hash.FileWrapper(rp_new.open("rb")))
+    except OSError:
```

Comment:
In any case, wouldn't a `finally` be a bit safer (i.e. more robust) here?

Reply:
No, because in success case, rp_signature would have been closed by DeltaFile and can't be closed a 2nd time (it can be checked but then it becomes quickly ugly). As said, lot of re factoring while keeping compatibility etc.

Reply:
I see, from the name I thought `close_if_necessary()` was idempotent.

I also understand the need for minimal changes to the current structure of the code base.
I first thought of something like this:

```python
    rp_new_fw = hash.FileWrapper(rp_new.open("rb"))
    rp_signature_fp = rp_signature.open("rb")
    return librsync.DeltaFile(rp_signature_fp, rp_new_fw))
```

to induce minimal behavioral change while addressing similar potential bugs where `rp_new.open()` would throw another type of unexpected exception.

But I guess `FileWrapper` wouldn't automatically close `rp_new` if not registered to the new `DeltaFile`, so we'd create the mirror bug where `rp_signature.open()` throws and `rp_new` doesn't get closed ¯\\_(ツ)_/¯